templates/base/server/.hathora/store.ts.hbs

```diff
@@ -103,14 +103,17 @@ const store: Application = {
       }
     }
     sendSnapshot(roomId, userId);
+    const { impl, chance } = rooms.get(roomId)!;
+    impl._onUserJoin(userId, ctx(chance, Date.now(), roomId));
```

Comment:
does this work if `impl._onUserJoin(...)` is undefined? should it be `impl?._onUserJoin(...)`

Reply:
_onUserJoin is an internal function and should always be defined, and its job may be a no-op if the user's function is not defined.